components/esp_wifi_remote/scripts/README.md

```diff
@@ -0,0 +1,38 @@
+# Parsing and generating code
+
+This folder contains scripts for parsing `esp_wifi` component headers and configs and generates configs, headers and sources of `esp_wifi_remote`.
+
+The CI job runs the parse and generation step, after which the generated files appear locally in this folder. Then the CI jobs compares the generated files with the specific version files. In case of a mismatch (CI failure), please check if the generated file is correct. If so, update the component and create a new version. If not, update the scripts.
+
+## Parsing `esp_wifi` headers
+
+Extract prototypes from the selected wifi headers
+
+## Generating `Kconfig`
+
+* Kconfig -- we only replace `SOC_WIFI_` -> `SLAVE_SOC_WIFI_` and `IDF_TARGET_` -> `SLAVE_IDF_TARGET_` configs
+* Kconfig.soc_wifi_caps.in -- we parse `IDF/components/soc/$SLAVE/include/soc/Kconfig.soc_caps.in` for all slaves and collect `SOC_WIFI_` configs, which we enclose with `if SLAVE_IDF_TARGET...` ... `endif`.
+This way the slave's WiFi capabilities become runtime available in `esp_wifi_remote` submenu
+
+## Generating headers and forwarding to `esp_hosted`
+
+Based on extracted WiFi function prototypes we generate
+
+* `esp_wifi_remote_api.h` -- declares all wifi APIs with `esp_wifi_remote...()` prefix
```

Comment:
I think there should be comment in all generated files that "they were autogenerated".

Reply:
will add a comment that the files are autogenerated

Reply:
All auto-generated files should have this comment now